staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/rules/rules.go

```diff
@@ -50,6 +53,22 @@ func exactOrWildcard(items []string, requested string) bool {
 	return false
 }
 
+var namespaceResource = schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"}
+
+func (r *Matcher) scope() bool {
+	if r.Rule.Scope == nil || *r.Rule.Scope == v1beta1.AllScopes {
+		return true
+	}
+	switch *r.Rule.Scope {
+	case v1beta1.NamespacedScope:
+		return r.Attr.GetResource() != namespaceResource && r.Attr.GetNamespace() != metav1.NamespaceNone
+	case v1beta1.ClusterScope:
+		return r.Attr.GetResource() == namespaceResource || r.Attr.GetNamespace() == metav1.NamespaceNone
```

Comment:
deserves a bit more comments I think - nuanced code.  fine to put it on the method.